### PR TITLE
Added joi validation on /login route

### LINKED ISSUE #4: Set up request body validation on /login route using Joi
Implement request body validation on the /login route using Joi. Ensure fields like email and password are validated before processing further.

**Reference:**
- [Joi Validation](https://www.npmjs.com/package/express-joi-validation)

### PR DESCRIPTION
**Pull request by Utkarsh-2005**
Added joi validation on /login route

Fixes #4 

### Type of Change
Enhancement

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other (please specify):

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

#### Additional Information 
- Name: Utkarsh Jha
- Email: 2206309@kiit.ac.in
- Roll No: 2206309
- Branch: IT

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
 import User from "../models/User.js";
 import bcrypt from "bcryptjs";
 import { createToken, verifyToken } from "../utils/jwtHelper.js";
+const Joi = require('joi')
```
Comment (ahsan-javaid) [inline controllers/auth.js:4]:
```suggestion
import Joi from 'joi';
```

File: controllers/auth.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 
-    if (!email || !password) {
-      return res
-        .status(400)
-        .json({ message: "Email and password are required" });
+    const schema = Joi.object().keys({
+      email: Joi.string().email().required(),
+      password: Joi.string().min(8).required(),
+    });
+    const { error } = schema.validate({ email, password });
```
Comment (ahsan-javaid) [inline controllers/auth.js:14]: Apply this validation as middleware on login route.
Example: https://github.com/MLSAKIIT/devblogs-backend/blob/96a3ef9b4124bfbe9d58ef2b9dacf101d5ad1613/routes/auth.js#L11

File: controllers/auth.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
 
 export const loginHandler = async (req, res) => {
   try {
     const { email, password } = req.body;
 
-    if (!email || !password) {
-      return res
-        .status(400)
-        .json({ message: "Email and password are required" });
+    const schema = Joi.object().keys({
```
Comment (ahsan-javaid) [inline controllers/auth.js:10]: Move this schema inside validators directory here: https://github.com/MLSAKIIT/devblogs-backend/tree/main/validators

Example: https://github.com/MLSAKIIT/devblogs-backend/blob/main/validators/register.js

**Review by ahsan-javaid — CHANGES_REQUESTED** [review]

File: controllers/auth.js (hunk 3 above)
Comment (Utkarsh-2005) [reply controllers/auth.js:10]: Thanks for your suggestion

**Comment by Utkarsh-2005** [comment]
@dead8309 kindly look at and merge these commit, as they fix the current issue as well as some previous errors

**Pull request merged**